### PR TITLE
🚧Improve release workflow

### LINKED ISSUE #105: 🚧Improve release workflow
### Complete The Item Below

- [X] I have updated the title without removing the 🚧 emoji.

### Description

Improve the release workflow by having it point to the reusable release workflow for dotnet GitHub actions.

This reusable workflow exists in the Infrastructure repository with the name _**dotnet-action-release.yml**_.

> **Note** There is a dependency on the reusable workflow being created first.  This should be shown below.

### Acceptance Criteria

- [x] Release workflow changed to use the reusable workflow below
- [x] The workflow will be setup to do preview OR production releases

### ToDo Items

- [X] Change type labels added to this issue.  Refer to the _**Change Type Labels**_ section below.
- [X] Priority label added to this issue.  Refer to the _**Priority Type Labels**_ section below.
- [X] Issue linked to the correct project _(if applicable)_.
- [X] Issue linked to the correct milestone _(if applicable)_.
- [x] Draft pull request created and linked to this issue _(only required with code changes)_.

### Issue Dependencies

- KinsonDigital/Infrastructure#80

### Related Work

_No response_

### Additional Information:

**_<details closed><summary>Change Type Labels</summary>_**

| Change Type         | Label                |
|---------------------|----------------------|
| Bug Fixes           | `🐛bug` |
| Breaking Changes    | `🧨breaking changes` |
| New Feature         | `✨new feature` |
| CICD Changes        | `♻️cicd` |
| Code Doc Changes    | `🗒️documentation/code` |
| Product Doc Changes | `📝documentation/product` |
</details>

**_<details closed><summary>Priority Type Labels</summary>_**

| Priority Type       | Label             |
|---------------------|-------------------|
| Low Priority        | `low priority`    |
| Medium Priority     | `medium priority` |
| High Priority       | `high priority`   |
</details>

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### PR DESCRIPTION
**Pull request by kselena**
<h1 style="font-weight:bold" align="center">Feature Pull Request</h1>
<h2 style="font-weight:bold" align="center">📋Review Checklist📋</h2>

- [x] The **_head(source)_** branch for this pull request is a **_feature_** branch, with the correct naming syntax, in the following example:
  - A **_head(source)_** branch is the branch being merged into a target branch.
    <details closed><summary>Example</summary>

      ``` xml
      Syntax: feature/<issue-num>-<description>
      Example: feature/123-my-feature
      ```
    </details>

- [x] The **_head(source)_** branch for this pull request is created from a **_development_** branch with the name **_develop_**.

- [x] The **_base(target)_** branch for this pull request is a **_development_** branch with the name **_develop_**.
  - A **_base(target)_** branch is the branch that the **_head(source)_** branch is merging into.

💡For more information on branching, refer to the project's [branching documentation](https://github.com/KinsonDigital/.github/blob/master/docs/BranchingFullGitFlow.md).

- [x] The associated issue exists and is linked to this pull request.
  - One issue per pull request.

- [x] The pull request title matches the title of the associated issue.

- [x] The labels attached to this PR match the labels attached to the associated issue.

- [x] The assigned project matches the project of the associated issue.

- [x] The assigned milestone matches the milestone of the associated issue.

- [x] My code follows the coding style of this project.
  - The style is enforced by the **_.editorconfig_** files in the project and displayed as warnings.  If there is an edge case, reach out, and let's discuss it.

- [x] I have manually tested my code changes to the best of my ability.

- [x] All tests passed locally.
  - This is required because unit tests might pass locally but not in the CI environment during the status check process or vice-versa.
  - Tests might pass on the developer's machine but not necessarily on the code reviewer's machine.
  - Status checks run for this pull request when merging into the **_develop_** branch.  These status checks run every time a change is pushed to the pull request.  These checks validate version syntax, tagging, builds, unit tests, and more.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -7,103 +8,63 @@ defaults:
...
 
 env:
-  project-name: BranchValidator
+  runs-on: ubuntu-latest
 
 
 on:
   workflow_dispatch:
     inputs:
       release_type:
```
Comment (KinsonDigitalAdmin) [inline .github/workflows/release.yml]: Change this workflow input to:

```suggestion
      release-type:
```

Make sure that all references to this input are updated as well

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -7,103 +8,63 @@ defaults:
...
+    needs: determine_build_config
+    uses: KinsonDigital/Infrastructure/.github/workflows/dotnet-action-release.yml@v8.1.0
+    with:
+      project-name: "${{ vars.PROJECT_NAME }}"
+      release-type: "${{ inputs.release_type }}"
+      run-branch: "${{ github.ref_name }}"
+      net-sdk-version: "${{ vars.NET_SDK_VERSION }}"
+      relative-release-notes-dir-path: "${{ vars.SCRIPT_RELATIVE_DIR_PATH }}"
+      release-notes-file-name-prefix: "${{ vars.RELEASE_NOTES_FILE_NAME_PREFIX }}"
+      build-config: "${{ needs.determine_build_config.outputs.build-config }}"
```
Comment (KinsonDigitalAdmin) [inline .github/workflows/release.yml]: Change this workflow input to:

```suggestion
      build-config: ${{ inputs.release-type == 'Production' && 'Release' || 'Debug' }}
```

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -7,103 +8,63 @@ defaults:
...
+            echo "Release build created.";
+          } else { # Any other branch than the release branch, do a debug build
+            "build-config=Debug" >> $env:GITHUB_OUTPUT;
+            echo "Debug build created.";
+          }
+
+
+  perform_release:
+    name: Performing ${{ inputs.release_type }} Release of ${{ vars.PROJECT_NAME }} (${{ needs.determine_build_config.outputs.build-config }})
+    needs: determine_build_config
```
Comment (KinsonDigitalAdmin) [inline .github/workflows/release.yml]: Remove:

```suggestion
```

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -7,103 +8,63 @@ defaults:
...
+          # If the branch that the workflow is running on is the required branch for the release, do a release build
+          if ( "${{ github.base_ref }}" -eq "main") {
+            "build-config=Release" >> $env:GITHUB_OUTPUT;
+            echo "Release build created.";
+          } else { # Any other branch than the release branch, do a debug build
+            "build-config=Debug" >> $env:GITHUB_OUTPUT;
+            echo "Debug build created.";
+          }
+
+
```
Comment (KinsonDigitalAdmin) [inline .github/workflows/release.yml]: Remove this job:

```suggestion
```

> **Note** Make sure that there are no empty lines between the `steps:` YAML key keyword and the `perform_release:` YAML key

**Review by KinsonDigitalAdmin — CHANGES_REQUESTED** [review]

**Review by KinsonDigitalAdmin — APPROVED** [review]
